vllm/core/scheduler.py

```diff
@@ -1160,6 +1165,10 @@ def free_finished_seq_groups(self) -> None:
                 self._finished_requests_ids.append(seq_group.request_id)
             else:
                 remaining.append(seq_group)
+            # Free finished seqs
```

Comment:
I think the problem here is that a sequence can be removed from self.running (not scheduled because it reaches max len) and then it won't be freed here. I think you can use a simple fix where you track the sequences that are finished inside the process_model_outputs (the lagged one) and then traverse the tracked sequences here and free them (independent of self.running)

Reply:
Resolved